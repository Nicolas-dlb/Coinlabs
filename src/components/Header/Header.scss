@import "./hamburgers/hamburgers.scss";
.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 2.5vh 3.2vh;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  z-index: 1010;
  width: 86%;
  .disable_event {
    pointer-events: none !important;
  }
  .btn_menu_active {
    fill: #787878 !important;
  }
  .header_input {
    display: none;
  }
  .menu_icon {
    position: relative;
    top: 3.5px;
  }

  .header_logo {
    display: flex;
    align-items: flex-end;
    justify-content: center;
    margin-left: 30px;
    cursor: pointer;
    .header_logo_container {
      display: flex;
      align-items: flex-end;
    }

    p {
      position: relative;
      top: 0.5vh;
      &::first-letter {
        font-size: 22px;
      }
    }

    .stick1 {
      border-radius: 20px;
      width: 3.5px;
      height: 20px;

      margin-right: 4px;
    }
    .stick2 {
      @extend .stick1;
      height: 25px;
    }
    .stick3 {
      @extend .stick1;
      height: 15px;
    }
    #logo_stick1 {
      background: #71ce98;
    }
    #logo_stick2 {
      background: #1baa63;
    }
    #logo_stick3 {
      background: #209654;
    }

    font-size: 20px;
    font-weight: bolder;
  }
  .header_right {
    display: flex;
    align-items: center;
    justify-content: flex-end;
    width: 60px;
    height: 20px;
    .header_user_icon,
    .header_user_name {
      display: none;
    }
    .green_dot {
      background: #16d366;
      border-radius: 50%;
      height: 6px;
      width: 6px;
      position: relative;
      left: -8px;
      bottom: 5px;
      pointer-events: none;
    }
    .header_notification {
      fill: var(--primary-font-color) !important;
      height: 20px;
      width: 20px;
      cursor: pointer;
    }

    #btn_search_mobile {
      fill: var(--primary-font-color);
      height: 25px;

      width: 25px;
      position: absolute;
      top: -12.5px;
      right: 9vw;

      pointer-events: none;
    }
  }
  img {
    position: relative;
    top: 2px;
    margin-left: 12px;
    height: 25px;
  }
}

.search_header {
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: fit-content;
  background-color: var(--app-bg);
}

label {
  position: relative;
  cursor: pointer;
}

.search-icon:before {
  content: "\f002";
  position: absolute;
  left: 16px;
  top: 16px;
  right: 16px;
  bottom: 0px;
  font-family: "FontAwesome";
  color: #828e9e;
}
.header_logout {
  position: absolute;
}
#search-bar {
  width: 16px;
  height: 16px;
  border: none;
  outline: none;
  padding: 16px;
  border-radius: 8px;
  font-size: 1.2em;
  font-family: inherit;
  transition: width 0.5s;
  position: absolute;
  top: -3vh;
  right: 5vw;
  color: var(--primary-font-color);
  background-color: var(--app-bg);
  caret-color: #1baa63;
  cursor: pointer;
}

#search-bar:focus {
  width: 50vw;
}

#search-bar:focus ~ .search-icon {
  transition: 0.1s;
  opacity: 0;
}

@media only screen and (max-width: 400px) {
  #search-bar:focus {
    width: 50vw;
    transition: 0.5s;
  }
}

@media only screen and (min-width: 992px) {
  .header {
    width: 95%;
  }
  .header_content {
    display: flex;
    align-items: center;
    justify-content: space-between;
    width: 28vw;
    .header_logo {
      position: relative;
      bottom: 5px;
      margin-left: 0;
    }
  }
  .search_header {
    display: none;
  }
  .search_icon_active {
    background: linear-gradient(
      146deg,
      rgba(11, 171, 80, 1) 0%,
      rgba(30, 176, 149, 1) 100%
    ) !important;
  }
  .search_icon {
    padding: 3px 0;
    border-radius: 4px;
    border-bottom-right-radius: 8px;
    border-top-right-radius: 8px;
    svg {
      position: relative;
      right: 5px;
      top: 2px;
    }
  }
  .header_input {
    display: flex !important;
    align-items: center;
    justify-content: space-between;
    border: 1px solid #2d293f;
    background: #1c182e;
    border-radius: 8px;
    padding-left: 10px;
    margin-left: 50px;

    input {
      border: none;
      outline: none;
      background: #1c182e;
      font-family: inherit;
      color: var(--primary-font-color);
      width: 180px;
      &::placeholder {
        font-family: inherit;
        color: #484558;

        font-size: 12px;
      }
    }
    svg {
      margin-left: 10px;
      height: 15px;
      width: 15px;
      padding: 0 5px;
      fill: var(--primary-font-color);
      cursor: pointer;
    }
  }
  .menu_icon {
    display: none;
  }

  .header_right {
    width: fit-content !important;
    position: relative;
    top: 5px;
    .header_right_search,
    .header_logout,
    #btn_search_mobile {
      display: none;
    }
    .header_user_icon,
    .header_user_name {
      display: inline-block !important;
    }
    .header_user_icon {
      border-radius: 8px;
      width: 27px;
      height: 27px !important;
      margin-bottom: 7px;
    }

    .header_user_name {
      margin: 0 5px;
      color: var(--secondary-font-color);
      display: flex !important;
      align-items: center;
      justify-content: center;
      cursor: pointer !important;
      padding: 5px;
      background: var(--app-bg);
      min-width: 60px;
      .username:hover {
        color: var(--primary-font-color);
      }

      &:hover {
        .header_user_logout {
          span:hover {
            color: var(--primary-font-color);
          }
        }
      }

      .header_user_logout_active {
        bottom: -17px !important;
      }
      .header_arrow {
        fill: var(--secondary-font-color);
        width: 10px;
        height: 10px;
        margin-left: 8px;
      }
      .header_user_logout {
        position: absolute;
        width: 75px;
        right: 1vw;
        bottom: 10px;
        height: 2vh;
        color: var(--third-font-color);
        align-self: flex-start;
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        transition: all 0.3s ease;
        z-index: -1;
        border-radius: 10px;
      }
    }
  }
}
