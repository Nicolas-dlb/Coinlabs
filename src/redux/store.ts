import { configureStore } from "@reduxjs/toolkit";
import marketReducer from "./reducers/marketSlice";
import walletsReducer from "./reducers/walletsSlice";
import appReducer from "./reducers/appSlice";
import userReducer from "./reducers/userSlice";

export default configureStore({
  reducer: {
    market: marketReducer,
    wallets: walletsReducer,
    app: appReducer,
    user: userReducer,
  },
});
