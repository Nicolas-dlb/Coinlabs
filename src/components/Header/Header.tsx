import { useState } from 'react'
import './Header.scss';
import { auth } from 'firebaseConfig';
import { useSelector } from 'react-redux';
import { selectProfilPic, selectUserName } from 'redux/reducers/userSlice';

function Header() {
  const [menuActive, setMenuActive] = useState(false);
  const username = useSelector(selectUserName);
  const profilPic = useSelector(selectProfilPic);
    /*<div id="menu_stick1" className="stick1"></div>
                <div id="menu_stick2" className="stick2"></div>
                <div id="menu_stick3" className="stick1"></div> */

     
        const logoutColor = menuActive ? "#787878" : "#FFF";
    return (
        <div className="header">
            <div onClick={() => {
                document.getElementById('toggleMenu')!.classList.toggle("is-active");
                document.getElementById("menu")!.classList.toggle('toggle_menu');
                document.getElementById('container')!.classList.toggle("container_active");
                
                document.getElementById('shadow')!.classList.toggle("shadow_active");
                document.getElementById('shadow')!.classList.toggle("opacity");
                setMenuActive(!menuActive);
               const bgColor = getComputedStyle(document.documentElement).getPropertyValue('--secondary-bg-color');
               if (bgColor !== "#221e34") {
document.documentElement.style.setProperty(
        "--secondary-bg-color",
        "#221e34"
      );       
      document.documentElement.style.setProperty(
        "--third-bg-color",
        "#29253b"
      );           
               } else {
                   document.documentElement.style.setProperty(
        "--secondary-bg-color",
        "#171429"
      );       
      document.documentElement.style.setProperty(
        "--third-bg-color",
        "#221e34"
      );           
               }
               
            }} className="menu_icon">
                <button id="toggleMenu" className="hamburger hamburger--squeeze" type="button">
  <span className="hamburger-box">
    <span className="hamburger-inner"></span>
  </span>
</button>
                
            </div>
            <div className="header_content">
              <div className="header_logo">
                <div className="header_logo_container">
                    <div id="logo_stick1" className="stick1"></div>
                    <div id="logo_stick2" className="stick2"></div>
                    <div id="logo_stick3" className="stick3"></div>
                </div>
                <p>Coinlabs</p>
            </div>
            
            </div>
            <div className="header_right">
               
                
                 <svg onClick={() => { 
                   document.getElementById('toggleMenu')!.click();
                   auth.signOut()
                 }} className="header_logout" height="20px" fill={logoutColor} xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><path d="M255.2 468.6H63.8c-11.7 0-21.3-9.5-21.3-21.3V64.6c0-11.7 9.5-21.3 21.3-21.3H255.2c11.8 0 21.3-9.5 21.3-21.3S266.9 0.9 255.2 0.9H63.8C28.6 0.9 0 29.5 0 64.6v382.7c0 35.2 28.6 63.8 63.8 63.8H255.2c11.8 0 21.3-9.5 21.3-21.3C276.4 478.1 266.9 468.6 255.2 468.6z"/><path d="M505.7 240.9L376.4 113.3c-8.3-8.2-21.8-8.1-30.1 0.2s-8.2 21.8 0.2 30.1l92.4 91.2H191.4c-11.8 0-21.3 9.5-21.3 21.3 0 11.8 9.5 21.3 21.3 21.3h247.6l-92.4 91.2c-8.4 8.3-8.4 21.7-0.2 30.1 4.2 4.2 9.7 6.3 15.1 6.3 5.4 0 10.8-2 14.9-6.1l129.3-127.6c4-4 6.3-9.4 6.3-15.1C512 250.3 509.7 244.9 505.7 240.9z"/></svg>
        
         
                 
                  <img className="header_user_icon"  src={
                    profilPic ||
                    "https://www.vhv.rs/dpng/d/164-1645859_selfie-clipart-groucho-glass-good-profile-hd-png.png"
                  } alt="" />
                
                <div onClick={() => {
                  document.getElementById('header_user_logout')!.classList.toggle('header_user_active')
                }} className="header_user_name">
                   <p className="username">{username}</p>
                   <svg className="header_arrow" xmlns="http://www.w3.org/2000/svg" width="452" height="452" viewBox="0 0 451.8 451.8"><path d="M225.9 354.7c-8.1 0-16.2-3.1-22.4-9.3L9.3 151.2c-12.4-12.4-12.4-32.4 0-44.8 12.4-12.4 32.4-12.4 44.7 0l171.9 171.9 171.9-171.9c12.4-12.4 32.4-12.4 44.7 0 12.4 12.4 12.4 32.4 0 44.8L248.3 345.4C242.1 351.6 234 354.7 225.9 354.7z"/></svg>
                <div id="header_user_logout" className="header_user_logout">
                  <p onClick={() => auth.signOut()}>Logout</p>
                </div>
                </div>
                
            </div>
        </div>
    )
}

export default Header
