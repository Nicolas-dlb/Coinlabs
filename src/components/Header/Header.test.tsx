import React from "react";
import { Provider } from "react-redux";
import { mount } from "enzyme";
import store from "redux/store";
import { act } from "react-dom/test-utils";
import Header from "./Header";

describe("Header", () => {
  let HeaderComponent: any;

  beforeEach(async () => {
    await act(async () => {
      HeaderComponent = await mount(
        <Provider store={store}>
          <Header />
        </Provider>
      );
    });
  });
  afterEach(() => {
    HeaderComponent = null;
  });

  it("should render correctly", () => {
    expect(HeaderComponent).toMatchSnapshot();
  });

  it("should disconnect user when click on logout", () => {
    const logoutComponent = HeaderComponent.find("#btn-logout").first();
    logoutComponent.simulate("click");
    expect(store.getState().user.userName).toBeNull();
  });
});
