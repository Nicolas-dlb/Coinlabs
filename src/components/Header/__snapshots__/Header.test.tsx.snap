// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Header>
    <div
      className="header"
    >
      <div
        className="menu_icon"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <button
          className="hamburger hamburger--squeeze"
          id="toggleMenu"
          type="button"
        >
          <span
            className="hamburger-box"
          >
            <span
              className="hamburger-inner"
            />
          </span>
        </button>
      </div>
      <div
        className="header_content"
      >
        <div
          className="header_logo"
          id="header_logo"
          onMouseEnter={[Function]}
        >
          <div
            className="header_logo_container"
          >
            <div
              className="stick1"
              id="logo_stick1"
            />
            <div
              className="stick2"
              id="logo_stick2"
            />
            <div
              className="stick3"
              id="logo_stick3"
            />
          </div>
          <p>
            Coinlabs
          </p>
        </div>
        <div
          className="header_input"
        >
          <input
            autoComplete="off"
            id="input_search"
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Search for crypto here"
            spellCheck="false"
            type="text"
          />
          <div
            className="search_icon"
          >
            <svg
              fill="#000000"
              height="50px"
              id="btn_search"
              onClick={[Function]}
              viewBox="0 0 50 50"
              width="50px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M 21 3 C 11.601563 3 4 10.601563 4 20 C 4 29.398438 11.601563 37 21 37 C 24.355469 37 27.460938 36.015625 30.09375 34.34375 L 42.375 46.625 L 46.625 42.375 L 34.5 30.28125 C 36.679688 27.421875 38 23.878906 38 20 C 38 10.601563 30.398438 3 21 3 Z M 21 7 C 28.199219 7 34 12.800781 34 20 C 34 27.199219 28.199219 33 21 33 C 13.800781 33 8 27.199219 8 20 C 8 12.800781 13.800781 7 21 7 Z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="header_right"
      >
        <div
          className="search_header"
          id="search_header"
        >
          <label
            htmlFor="search"
          >
            <input
              id="search-bar"
              onBlur={[Function]}
              onClick={[Function]}
              onKeyDown={[Function]}
              type="text"
            />
            <svg
              fill="#000000"
              height="50px"
              id="btn_search_mobile"
              viewBox="0 0 50 50"
              width="50px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M 21 3 C 11.601563 3 4 10.601563 4 20 C 4 29.398438 11.601563 37 21 37 C 24.355469 37 27.460938 36.015625 30.09375 34.34375 L 42.375 46.625 L 46.625 42.375 L 34.5 30.28125 C 36.679688 27.421875 38 23.878906 38 20 C 38 10.601563 30.398438 3 21 3 Z M 21 7 C 28.199219 7 34 12.800781 34 20 C 34 27.199219 28.199219 33 21 33 C 13.800781 33 8 27.199219 8 20 C 8 12.800781 13.800781 7 21 7 Z"
              />
            </svg>
          </label>
        </div>
        <svg
          className="header_logout"
          fill="#FFF"
          height="20px"
          id="header_logout"
          onClick={[Function]}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M255.2 468.6H63.8c-11.7 0-21.3-9.5-21.3-21.3V64.6c0-11.7 9.5-21.3 21.3-21.3H255.2c11.8 0 21.3-9.5 21.3-21.3S266.9 0.9 255.2 0.9H63.8C28.6 0.9 0 29.5 0 64.6v382.7c0 35.2 28.6 63.8 63.8 63.8H255.2c11.8 0 21.3-9.5 21.3-21.3C276.4 478.1 266.9 468.6 255.2 468.6z"
          />
          <path
            d="M505.7 240.9L376.4 113.3c-8.3-8.2-21.8-8.1-30.1 0.2s-8.2 21.8 0.2 30.1l92.4 91.2H191.4c-11.8 0-21.3 9.5-21.3 21.3 0 11.8 9.5 21.3 21.3 21.3h247.6l-92.4 91.2c-8.4 8.3-8.4 21.7-0.2 30.1 4.2 4.2 9.7 6.3 15.1 6.3 5.4 0 10.8-2 14.9-6.1l129.3-127.6c4-4 6.3-9.4 6.3-15.1C512 250.3 509.7 244.9 505.7 240.9z"
          />
        </svg>
        <div
          className="header_notification"
        >
          <div>
            <svg
              className="not"
              fill="white"
              id="notification"
              onClick={[Function]}
              viewBox="0 5.4 32 32"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                data-name="Layer 31"
                id="Layer_31"
              >
                <path
                  d="M27,27H5a1,1,0,0,1-.89-1.45,18.14,18.14,0,0,0,1.89-8V14a10,10,0,0,1,20,0v3.53a18.14,18.14,0,0,0,1.89,8A1,1,0,0,1,27,27ZM6.55,25h18.9A20.14,20.14,0,0,1,24,17.53V14A8,8,0,0,0,8,14v3.53A20.14,20.14,0,0,1,6.55,25Z"
                />
                <path
                  d="M16,31a5,5,0,0,1-5-5,1,1,0,0,1,2,0,3,3,0,0,0,.88,2.12,3.08,3.08,0,0,0,4.24,0,1,1,0,0,1,1.42,1.42A5,5,0,0,1,16,31Z"
                />
                <path
                  d="M16,6a1,1,0,0,1-1-1V2a1,1,0,0,1,2,0V5A1,1,0,0,1,16,6Z"
                />
                <path
                  d="M26,5a2,2,0,1,1,2-2A2,2,0,0,1,26,5Zm0-2h0Zm0,0h0Zm0,0h0Zm0,0h0Zm0,0h0Zm0,0h0Zm0,0h0Zm0,0h0Z"
                />
              </g>
            </svg>
          </div>
        </div>
        <img
          alt=""
          className="header_user_icon"
          src={
            Object {
              "canInstrument": true,
              "createTransformer": [Function],
              "getCacheKey": [Function],
              "process": [Function],
            }
          }
        />
        <div
          className="header_user_name log"
          id="header_user_name"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <p
            className="username log"
            onFocus={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
          />
          <svg
            className="header_arrow log"
            height="452"
            id="header_arrow"
            viewBox="0 0 451.8 451.8"
            width="452"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M225.9 354.7c-8.1 0-16.2-3.1-22.4-9.3L9.3 151.2c-12.4-12.4-12.4-32.4 0-44.8 12.4-12.4 32.4-12.4 44.7 0l171.9 171.9 171.9-171.9c12.4-12.4 32.4-12.4 44.7 0 12.4 12.4 12.4 32.4 0 44.8L248.3 345.4C242.1 351.6 234 354.7 225.9 354.7z"
            />
          </svg>
          <div
            className="header_user_logout log"
            id="header_user_logout"
          >
            <span
              className="log"
              id="btn-logout"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              Logout
            </span>
          </div>
        </div>
      </div>
    </div>
  </Header>
</Provider>
`;
